src/app/components/curriculum-creator-page/curriculum-creator-page.component.spec.ts

```diff
@@ -51,19 +59,556 @@ describe('CurriculumCreatorPageComponent', () => {
+  });
+});
+
+describe('NewCurriculumDialogComponent', () => {
+  let component: NewCurriculumDialog;
+  let fixture: ComponentFixture<NewCurriculumDialog>;
+  let toastrService:ToastrService;
+  let curriculum:Curriculum;
+  const mockDialogRef = {
+    close: jasmine.createSpy('close')
+  }
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [ NewCurriculumDialog ],
+      providers:[
+        { provide: MAT_DIALOG_DATA, useValue: {name: ''} },
+        { provide: MatDialogRef, useValue: mockDialogRef},
+        ToastrService
+      ],
+      imports: [
+        MatProgressSpinnerModule,
+        MatCardModule,
+        DragDropModule,
+        MatExpansionModule,
+        MatFormFieldModule,
+        MatInputModule,
+        MatIconModule,
+        MatToolbarModule,
+        MatDialogModule,
+        MatGridListModule,
+        MatCheckboxModule,
+        MatMenuModule,
+        FormsModule,
+        MatTooltipModule,
+        ToastrModule.forRoot(),
+        HttpClientTestingModule,
+        BrowserAnimationsModule,
+      ]
+    })
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(NewCurriculumDialog);
+    component = fixture.componentInstance;
+    toastrService = TestBed.get(ToastrService);
+    fixture.detectChanges();
+    component.cs.nodes = [];
+    component.cs.idToCurriculum = new Map();
+    component.data.name = 'Test';
+    curriculum = new Curriculum(1,"Java",[]);
+  });
+
+  it('Should create the component', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('Should test onNoClick', () => {
+    component.onNoClick();
+    expect(mockDialogRef.close).toHaveBeenCalled();
+  });
+
+  it('Should test submit, error 1', () => {
+    component.data.name = undefined;
+    component.submit();
+    expect(toastrService.previousToastMessage).toBe('Please fill in the input field!');
+  });
+
+   it('Should test submit, success', () => {
+    spyOn(component.cfs, 'createCurriculum').and.returnValue(of(curriculum));
+    component.submit();
+    expect(toastrService.previousToastMessage).toBe('Curriculum successfully made');
+  });
+
+  it('Should test submit, toastr error 2', () => {
+    spyOn(component.cfs,'createCurriculum').and.returnValue(of(null))
+    component.submit();
+    expect(toastrService.previousToastMessage).toBe('There was an error creating the Curriculum');
+  });
+
+  it('Should test submit, toastr error 3', () => {
+    const observable: Observable<Curriculum> = new Observable<Curriculum>((observer) => {
+      observer.error({status: 400, statusText: "Bad Request"})
+      observer.complete();
+    });
+    spyOn(component.cfs,'createCurriculum').and.returnValue(observable);
+    component.submit();
+    expect(toastrService.previousToastMessage).toBe('Failed to communicate with the server');
+  });
+
+});
+
+describe('DeleteCurriculumDialog', () => {
```

Comment:
There should perhaps be comments explaining the goal of each group of tests at the corresponding describe. Further explanation can be provided at the beforeEach, as well as each test as necessary. Preferably at least the first test should have a bit of an explanation, and if the following tests are similar enough, they don't need one. But at least one comment for the first test explaining what is happening.